sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java

```diff
@@ -431,13 +425,19 @@ public void run() {
         return future;
     }
 
-    private synchronized void instanceSignOut() {
-        if (mTokenStorage.getToken() == null) {
-            AppCenterLog.warn(LOG_TAG, "Couldn't sign out: authToken doesn't exist.");
-            return;
-        }
-        removeTokenAndAccount();
-        AppCenterLog.info(LOG_TAG, "User sign-out succeeded.");
+    private void instanceSignOut() {
+        post(new Runnable() {
+
+            @Override
+            public void run() {
+                if (mTokenStorage.getToken() == null) {
+                    AppCenterLog.error(LOG_TAG, "Cannot sign out because not signed in.");
```

Comment:
Seems a warning, not an error. What about *A user hasn't signed in, skip sign-out"?

Reply:
It's an error, its invalid usage of the SDK to call signOut if not signed in.

Reply:
Used a different wording.